gh-92859: Doc: add info about logging.debug() calling basicConfig() (GH-93063)

Gaasmann opened the pull request:
This adds some info about logging.debug() (and other level functions)
calling basicConfig() on behalf of the user if logging wasn't configured
beforehand.

#92859 
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->

vsajip commented on Doc/library/logging.rst:
```diff
@@ -242,6 +242,10 @@ is the module's name in the Python package namespace.
       above example). In such circumstances, it is likely that specialized
       :class:`Formatter`\ s would be used with particular :class:`Handler`\ s.
 
+      If no handler is attached to this logger (or its parents if
```
Perhaps change to
```
If no handler is attached to this logger (or any of its ancestors,
taking into account the relevant :attr:`Logger.propagate` attributes),
the message will be sent ...
```

Gaasmann replied:
Addressed on d173e5b75404a6dbf8883df24928f53795cef2ad

vsajip commented on Doc/library/logging.rst:
```diff
@@ -1038,6 +1042,9 @@ functions.
    above example). In such circumstances, it is likely that specialized
    :class:`Formatter`\ s would be used with particular :class:`Handler`\ s.
 
+   The function will call :func:`basicConfig` if the root logger doesn't have
```
Change to
```
This function (as well as :func:`info`, :func:`warning`, :func:`error` and :func:`critical`) will call ...
```

Gaasmann replied:
Addressed on d173e5b75404a6dbf8883df24928f53795cef2ad

vsajip reviewed (changes requested):
Thanks, I've asked for some minor tweaks.

Gaasmann committed:
Doc: add info about logging.debug() calling basicConfig()

This adds some info about logging.debug() (and other level functions)
calling basicConfig() on behalf of the user if logging wasn't configured
beforehand.

Gaasmann committed:
Address comments from @vsajip

Gaasmann reviewed (commented):
I have made the requested changes; please review again
(sorry for the force push, I reset my email on the first commit)
